File: src/com/lucas/jann/javascript/main/Layer.js
import Neuron from './Neuron.js';


export default class Layer {

    constructor(numNeurons, numPrevious) {
        this.numNeurons = numNeurons;
        this.neurons = [];
        for (let i = 0; i < numNeurons; i++) {
            if (numPrevious === 0) {
                this.neurons.push(new Neuron(numNeurons));
            } else {
                this.neurons.push(new Neuron(numPrevious));
            }
        }
    }

    neuronsActivations(inputs) {
        const activations = [];
        for (let i = 0; i < this.numNeurons; i++) {
            activations.push(this.neurons[i].activation(inputs));
        }
        return activations;
    }
}

File: src/com/lucas/jann/java/main/Layer.java
package com.lucas.jann.java.main;

class Layer {

    int numNeurons;
    private Neuron[] neurons;

    Layer(int numNeurons, int numPrevious) {
        this.numNeurons = numNeurons;
        neurons = new Neuron[numNeurons];

        for (int i = 0; i < numNeurons; i++) {
            if (numPrevious == 0) {
                neurons[i] = new Neuron(numNeurons);
            } else {
                neurons[i] = new Neuron(numPrevious);
            }
        }
    }

    double[] neuronsActivations(double... inputs) {
        double[] activations = new double[numNeurons];
        for (int i = 0; i < numNeurons; i++) {
            activations[i] = neurons[i].activation(inputs);
        }
        return activations;
    }
}
